code/fcn_2D_segm/train_fcn8.py

```diff
@@ -0,0 +1,398 @@
+#!/usr/bin/env python2
+
+import os
+import argparse
+import time
+from getpass import getuser
+from distutils.dir_util import copy_tree
+
+import numpy as np
+import theano
+import theano.tensor as T
+from theano import config
+import lasagne
+from lasagne.regularization import regularize_network_params
+
+from data_loader import load_data
+from fcn8 import buildFCN8
+# from metrics import jaccard, accuracy, crossentropy
+
+
+_FLOATX = config.floatX
+_EPSILON = 10e-8
+
+
+def jaccard_metric(y_pred, y_true, n_classes, one_hot=False):
+
+    assert (y_pred.ndim == 2) or (y_pred.ndim == 1)
+
+    # y_pred to indices
+    if y_pred.ndim == 2:
+        y_pred = T.argmax(y_pred, axis=1)
+
+    if one_hot:
+        y_true = T.argmax(y_true, axis=1)
+
+    # Compute confusion matrix
+    cm = T.zeros((n_classes, n_classes))
+    for i in range(n_classes):
+        for j in range(n_classes):
+            cm = T.set_subtensor(
+                cm[i, j], T.sum(T.eq(y_pred, i) * T.eq(y_true, j)))
```

Comment:
Also, you seem to redefine that in cnn_1D_segm/train_fcn1D.py
Maybe it would be a good idea to share these kind of functions between the 3 segmentation models.